Add documentation for new Rainforest Eagle-200 platform

gtdiehl opened the pull request:
**Description:**
New documentation for Rainforest Eagle-200 platform

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#24919
## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html


<a href="https://gitpod.io/#https://github.com/home-assistant/home-assistant.io/pull/9761"><img src="https://gitpod.io/api/apps/github/pbs/github.com/gtdiehl/home-assistant.io.git/5d9523241db74c2c86220a1e677c9cafeb3e46b0.svg" /></a>



<a href="https://gitpod.io/#https://github.com/home-assistant/home-assistant.io/pull/9761"><img src="https://gitpod.io/api/apps/github/pbs/github.com/gtdiehl/home-assistant.io.git/f0b0ab2334c53ee479a1aca5eef0896df98be447.svg" /></a>

gtdiehl committed:
Add doc for new component Rainforest Eagle-200

gtdiehl committed:
Added Rainforest Logo and Examples to documentation

klaasnicolaas commented on source/_components/rainforest_eagle.markdown:
```diff
@@ -0,0 +1,72 @@
+  - Energy
+ha_release: 0.96
+ha_iot_class: Local Polling
+redirect_from:
```
Please remove the `redirect_from`

gtdiehl replied:
Removed `redirect_from`

klaasnicolaas commented on source/_components/rainforest_eagle.markdown:
````diff
@@ -0,0 +1,72 @@
+```
+
+```yaml
+# example configuration.yaml entry, limiting the metrics to demand only
````
Move this example underneed the configuration variables and give it a full example title

gtdiehl replied:
Moved the example underneath the configuration variables and created an examples section with two examples.

klaasnicolaas commented on source/_components/rainforest_eagle.markdown:
```diff
@@ -0,0 +1,72 @@
+comments: false
+sharing: true
+footer: true
+ha_category:
```
Maybe add `sensor`?

gtdiehl replied:
Added `Sensor` to `ha_category`

gtdiehl committed:
Fixed requested changes

klaasnicolaas committed:
:pencil2: Tweak

klaasnicolaas commented:
✅ Can be merged after merging the parent PR 👍

gtdiehl committed:
Removed Examples and Variable selectors

gtdiehl committed:
Removed Examples and Variable selectors

gtdiehl commented:
I updated the documentation and removed the Examples and variable selectors as I had to remove direct access to them in the code per Architecture issue #100 [https://github.com/home-assistant/architecture/blob/master/adr/0003-monitor-condition-and-data-selectors.md](url)

gtdiehl committed:
Fixed merge issue

frenck committed:
:pencil2: Removal of obsolete code

frenck reviewed (approved):
Thanks, @gtdiehl! 👍
